docs/src/modules/components/DemoSandboxed.js

```diff
@@ -25,13 +27,25 @@ function FramedDemo(props) {
     };
   }, [document]);
 
+  const cache = React.useMemo(
+    () =>
+      createCache({
+        key: 'iframe-demo',
```

Comment:
It seems that we can share the same cache between two iframe demos. The documentation only mentions that unique keys are required when using the same window/frame https://emotion.sh/docs/@emotion/cache#key.

@Andarist is it accurate?

Reply:
I don't believe that you can share the same cache between iframes because each one has to get their separate styles in their separate documents. Cache only checks the inserted keys and assumes that inserted keys are in sync with the inserted style elements - but if you share cache between "realms" then styles will really be only inserted into a single document.

Reply:
Separate note - `useMemo` doesn't give any semantic guarantees about caching (it's just like a perf hint). This probably doesn't matter for you here but to avoid `useMemo` refiring (even if `document` doesn't change!) you should put cache into a ref or into a state.

Reply:
> Cache only checks the inserted keys and assumes that inserted keys are in sync with the inserted style elements

In the current configuration, each iframe has a different cache instance in the context with the same key. Is there a global side effect between different cache instances?

> useMemo doesn't give any semantic guarantees about caching

Would the style be broken if the emotion cache change between two renders?

Reply:
> Is there a global side effect between different cache instances? 

There is:

https://github.com/emotion-js/emotion/blob/6e157aa198ac32d326d1aa1a921f459d880a8428/packages/cache/src/index.js#L84-L99

But it operates on the main script's document so it should simply not find any iframe styles in it and this should act like a noop - like you have mentioned.

> Would the style be broken if the emotion cache changes between two renders? I ask because I'm happy with a perf hint only.

No - styles should not be affected. You would only end up with multiple copies of the same styles being injected and some perf hit caused by somewhat full subtree rerended (all context consumers would have to rerender)

Reply:
Ok great, awesome, thanks for the details